hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/CallIdSequence.java

```diff
@@ -131,30 +139,32 @@ public long getLastCallId() {
             return longs.get(INDEX_HEAD);
         }
 
-        long getTail() {
-            return longs.get(INDEX_TAIL);
-        }
-
         @Override
         public int getMaxConcurrentInvocations() {
             return maxConcurrentInvocations;
         }
 
         @Override
         public long next(Invocation invocation) {
-            assert invocation.op.getCallId() == 0 : "callId should be null:" + invocation;
+            final Operation op = invocation.op;
+            assert !invocation.isActive() : "Invocation is already in progress:" + invocation;
```

Comment:
this check seems redundant, since setCallId will already fail later on with an active invocation. The method can be then simplified as `long next(boolean isUrgent)`, furthermore, it's probably better to make it throw a checked exception and handle the exception at the caller site.